src/main/java/org/broadinstitute/dsm/DSMServer.java

```diff
@@ -668,4 +703,69 @@ public static boolean isTest(@NonNull String shipperName) {
         }
         return true;
     }
+
+    private static void registerAppEngineStartupCallback(long bootTimeoutSeconds) {
+        // Block until isReady is available, with an optional timeout to prevent
+        // instance for sitting around too long in a nonresponsive state.  There is a
+        // judgement call to be made here to allow for lengthy liquibase migrations during boot.
+        logger.info("Will wait for at most {} seconds for boot before GAE termination", bootTimeoutSeconds);
+        get("/_ah/start",new ReadinessRoute(bootTimeoutSeconds));
+    }
+
+    private static class ReadinessRoute implements Route {
+
+        private final long bootTimeoutSeconds;
+
+        public ReadinessRoute(long bootTimeoutSeconds) {
+            this.bootTimeoutSeconds = bootTimeoutSeconds;
+        }
+
+        @Override
+        public Object handle(Request req, Response res) throws Exception {
+            AtomicInteger status = new AtomicInteger(HttpStatus.SC_SERVICE_UNAVAILABLE);
+            long bootTime = Instant.now().toEpochMilli();
+            Thread waitForBoot = new Thread(() -> {
```

Comment:
@MocanaAtBroad @yufengwng this turned out to be a bit more complicated than I originally thought, but it brings up something that I think we need to address in study-server as well: letting an instance start fully (allowing for liquibase to complete, which can take a while) before GAE sends requests to the instance.  Based on [this doc](https://cloud.google.com/appengine/docs/standard/java/how-instances-are-managed#startup), I think the approach here (to have `ah/start` block until things are ready) should work.  Otherwise, the default `404`  that spark returns will tell GAE "everything is fine--please start sending traffic to this instance", which is probably premature, since even the shortest deployments seem to take ~10s, without liquibase doing much.

Reply:
Also, I'm really glad y'all mentioned the whole `/ah/start` thing.  It was driving me kinda bonkers.

Reply:
A sampling of more recent deploys shows cleaner `ah/start` calls and their duration.  Note the lack of `404`, `500`, or `503` at top.
<img width="741" alt="Screen Shot 2020-06-25 at 4 01 51 PM" src="https://user-images.githubusercontent.com/1653048/85789809-3aefda00-b6fd-11ea-94c7-23d031aa0ec7.png">